### ISSUE TITLE
[Bug]: Filters not restored from Query String

### ISSUE THREAD

**Issue by booni3** [rappasoft/laravel-livewire-tables#1965]
### What happened?

I expect the query string, both search and filters to re-apply when the page is refreshed or navigating back. Search applies as expected, but filters do not.

I am able to get filters to apply using the `storeFiltersInSessionEnabled()` config option.

Without this, the filters and query string are cleared on refresh.

I have tried to trace this issue, and am able to see in the following functions:

```php
    public function restoreFilterValues(): void
    {
        if (empty($this->filterComponents) || empty($this->appliedFilters)) {
            $this->filterComponents = $this->appliedFilters = $this->getStoredFilterValues();
        }
    }

    public function getStoredFilterValues(): array
    {
        if ($this->shouldStoreFiltersInSession() && session()->has($this->getFilterSessionKey())) {
            return session()->get($this->getFilterSessionKey());
        }

        return [];
    }
```

`getStoredFilterValues` returns [] when a query string is set. Therefore, with session storage disabled:

```php
    public function mountFilterHelpers(): void
    {
        dd($this->appliedFilters); // filters present
        
        $this->restoreFilterValues();

        dd($this->appliedFilters); // filters empty
    ....
```

### Package Version

3.4.20

### PHP Version

None

### Laravel Version

11.22

### Alpine Version

3.4

### Theme

None

**Comment by lrljoe** [comment]
This should be fixed in the latest version (3.4.21)

**Comment by lrljoe** [comment]
Closing, please re-open if issue persists!

**Comment by lrljoe** [comment]
I suspect this is a hangover from some v3 beta work, I'll validate this weekend and get a fix in place

**Comment by lrljoe** [comment]
Just to update on this, there are some imminent fixes in the next release to allow enabling/disabling the queryString independently for search/filters etc

This also fixes this issue.

**Comment by lrljoe** [comment]
This should be fixed in 3.5.0, please let me know if not the case!
